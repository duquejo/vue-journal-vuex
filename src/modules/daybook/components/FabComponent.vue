<template>
  <button class="btn btn-primary">
    <i class="fa fa-2xl" :class="icon"/>
  </button>
</template>

<script>
export default {
  props: {
    icon: {
      type: String,
      default: 'fa-plus',
    }
  },
}
</script>

<style lang="scss" scoped>
button {
  border-radius: 100%;
  border: 0px;
  bottom: 20px;
  height: 80px;
  position: fixed;
  right: 20px;
  width: 80px;

  &:hover {
    background-color: lighten($color: #2c3e50, $amount: 15);
  }
}
</style>