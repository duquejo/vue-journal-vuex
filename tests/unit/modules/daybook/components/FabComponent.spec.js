import { shallowMount } from "@vue/test-utils";
import FabComponent from '@/modules/daybook/components/FabComponent';

describe('FabComponent.vue', () => {
  it('Should match the snapshot', () => {
    const wrapper = shallowMount(FabComponent);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should show the default icon', () => {
    const wrapper = shallowMount(FabComponent);
    const icon = wrapper.find('i');

    expect(icon.classes('fa-plus')).toBeTruthy();
  });

  it('Should show the icon fa-circle if the args are been supplied', () => {
    const wrapper = shallowMount(FabComponent, {
      props: {
        icon: 'fa-circle',
      },
    });
    const icon = wrapper.find('i');
 
    expect(icon.classes('fa-circle')).toBeTruthy();
  });

  it('Should emit the on:click event if its clicked', () => {
    const wrapper = shallowMount(FabComponent);
    
    wrapper.find('button').trigger('click');

    expect(wrapper.emitted('on:click')).toHaveLength(1);
  });
});
