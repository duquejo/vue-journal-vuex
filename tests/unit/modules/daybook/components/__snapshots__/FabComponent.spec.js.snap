// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FabComponent.vue Should match the snapshot 1`] = `<button class="btn btn-primary"><i class="fa fa-2xl fa-plus"></i></button>`;
